BackwardConfig.cmake

```diff
@@ -216,7 +216,7 @@ if(WIN32)
 			set(CMAKE_EXE_LINKER_FLAGS "-Wl,--pdb= ")
 			add_compile_options(-gcodeview)
 		else()
-			set(MINGW_MSVCR_LIBRARY "msvcr90$<$<CONFIG:DEBUG>:d>" CACHE STRING "Mingw MSVC runtime import library")
+			find_library(MINGW_MSVCR_LIBRARY ucrtbase msvcr140 msvcr120 msvcr110 msvcr100 msvcr90 REQUIRED DOC "Mingw MSVC runtime import library")
```

Comment:
I'm not a fan of using `find_library` to find system libraries.
It requires a non-minimal cmake toolchain file that includes the paths of the system libraries.

The following fails:
```
CC=x86_64-w64-mingw32-gcc CXX=x86_64-w64-mingw32-g++ cmake -S . -B build -GNinja -DCMAKE_SYSTEM_NAME=Windows
(snip)
CMake Error at BackwardConfig.cmake:219 (find_library):
  Could not find MINGW_MSVCR_LIBRARY using the following names: ucrtbase,
  msvcr140, msvcr120, msvcr110, msvcr100, msvcr90
Call Stack (most recent call first):
  CMakeLists.txt:32 (include)
```
When I add the library paths of the mingw64 system libraries to `CMAKE_FIND_ROOT_PATH`, configuration succeeds.

Also, this does not fix `__imp__set_abort_behavior` (reported by me in https://github.com/bombela/backward-cpp/issues/291#issuecomment-1716335403).
I think it's better to fix the missing symbol error first before addressing this.

Reply:
I understand you're cross-compiling with mingw so it is normal that cmake is not able to resolve the libraries. It is expected that you [supply a toolchain file](https://cmake.org/cmake/help/book/mastering-cmake/chapter/Cross%20Compiling%20With%20CMake.html).

As reported in [#291](https://github.com/bombela/backward-cpp/issues/291#issuecomment-1906529173) the current master does not have any missing symbol.

Instead of `find_library` what would you recommend?

Reply:
It is true we are not interested in the msvcr90 location, we're interested finding out what C library is present on the system instead of fixing msvcr90 (visual studio 2008, uncommon nowadays).

Would you care to explain somehow why find_library and check_c_source_compiles are different in this case? I understand that both require compiler and linker.

Reply:
`find_library` is a cmake function that will look for a msvcr90.dll.a/msvcr90.a file (on msvc) or libmsvcr90.dll.a/libmsvcr90.a (with gcc). So it's important for CMake to know it what locations to search: this needs to be configured by the cmake toolchain.
With `check_c_source_compiles`, cmake just says to the library: try to compile+link this file and link `msvcr90.lib`/`-lmsvcr90`. The compiler itself will look in its preconfigured locations (cmake does not need to know about them).